index.rst

```diff
@@ -4,9 +4,184 @@ File Formats and Layouts for Cell-based Coadds
+
+Image Data Layouts
+==================
+
+This section will cover potential ways to lay out the image data - the 5+ planes that share a common grid, as well as the PSF images - across multiple FITS HDUs.
+
+Binary Table With Per-Cell Rows
+-------------------------------
+
+Since FITS binary tables can have array columns with arbitrary shapes, the most intuitive layout for cell-based image data is probably a single binary table HDU that has each image plane as a separate 2-d array column.
+A FITS WCS can be stored with each row, using "Green Bank" convention to translate header keys to table columns; this may be sufficient for third-party readers to fully interpret our files and even stitch together cells on the fly (in the case of image viewers).
+
+It's not clear whether any extant FITS viewers could *already* do this in an intuitive way, but by relying on an existing convention it'd be reasonable for them to add it, even if they were otherwise disinclined to add support for observatory-specific file formats.
+
+The Green Bank convention makes no mention of having multiple images as different columns in the same table, however, so we may want to consider putting different image planes in different binary table HDUs, each with corresponding rows, and probably some columns - like the WCS ones - duplicated.
+The extra storage cost of another header and some duplicated columns is insignificant, and it may be desirable to put multiple cells from the same plane closer together on disk than different planes of the same cell, especially since we expect that subimage reads of some planes (i.e. the image data image) will be much more popular than others (e.g. noise realizations).
+
+PSF images fit naturally in this layout, as there's nothing wrong with having some array columns having a different shape.
+There might be some confusion with a Green Bank convention WCS, though, if we go with a single table with different columns for different image planes as well as the PSF; there's no way to mark which image columns the WCS applies to.
+
+The main problem with binary table layouts is compression support.
+The latest FITS standard does include (in section 10.3) a discussion of "tiled table compression", which seems like it'd be sufficient, as long as compressing one cell at a time is enough to get a good compression ratio (this is unclear).
+Unlike image tile compression, binary table tile compression doesn't support lossy compression algorithms or dithered quantization, but it would still be possible to do our own non-dithered quantization and use ``BZERO`` and ``BSCALE`` to record the transformation from integer back to floating-point.
+The bigger problem is that there does not appear to be any implementations of it: there is no mention of it in either the CFITSIO or Astropy documentation (and even if an implementation does exist in, say, the Java ecosystem, we wouldn't be in a position to use it).
```

Comment:
It appears I was wrong about this: as @gpdf discovered, both CFITSIO and at least one major Java FITS implementation ([nom.tam.fits](https://github.com/nom-tam-fits/nom-tam-fits)) do support table compression (CFITSIO just hasn't updated their docs to reflect this).

But having looked more closely at the table compression spec, I'm less sanguine about introducing our own quantization convention for lossy compression here, for two reasons:

- Adding `BZERO` and `BSCALE` columns as per the Green Bank convention only works if there is only one image column in the binary table, i.e. we put each logical plane in its own binary table HDU (this seems generally true of the Green Bank convention: there's no provision for multiple image columns).  This is fine, but I think it removes one of the advantages of the binary table form.

- There'd be no way to record the "subtractive dithering" quantization (in which one adds a deterministic uniform random number before rounding to int, and subtracts it when decompressing).  And while we could invent one (probably a `ZDITHER0` column, by analogy with the tiled image compression header key), users would have to use our code to correctly decompress in that case.  While other file layouts would require users who don't use our code to do some work, none of them would require them to implement anything as low-level as decompression.  @beckermr, do you know if you used subtractive dithering in your lossy-compression configuration?  It'd be the `ZQUANTIZ` header key.

In fact, if we decide to do binary tables with one plane per HDU to work around the first problem, we could just make them the very binary tables used to implement tiled image compression, but add extra columns (which are explicitly permitted) for e.g. a Green Bank convention WCS or other metadata.  FITS libraries often provide the option to read a compressed image HDU as an image or as a binary table, and with this format we'd provide all information in a fairly FITS-native way in at least one of those two interpretations.  This does mean that the tiled image compression representation would correspond to the exploded image, of course.

Reply:
Se use SUBTRACTIVE_DITHER_2 which preserves zeros exactly. You need dithering to remove biases from the lossy compression and having zeros go through unchanged has lots of nice advantages.